### PR TITLE
Bump Jackson to 2.10.0

### PR DESCRIPTION
**Pull request by Fokko**
This fixes some CVE's:

**High severity vulnerability that affects com.fasterxml.jackson.core:jackson-databind**
An issue was discovered in FasterXML jackson-databind prior to 2.7.9.4, 2.8.11.2, and 2.9.6. When Default Typing is enabled (either globally or for a specific property), the service has the Jodd-db jar (for database access for the Jodd framework) in the classpath, and an attacker can provide an LDAP service to access, it is possible to make the service execute a malicious payload.

Affected versions: < 2.7.9.4

**High severity vulnerability that affects com.fasterxml.jackson.core:jackson-databind**
A deserialization flaw was discovered in the jackson-databind, versions before 2.6.7.1, 2.7.9.1 and 2.8.9, which could allow an unauthenticated user to perform code execution by sending the maliciously crafted input to the readValue method of the ObjectMapper.

Affected versions: < 2.6.7.1

**High severity vulnerability that affects com.fasterxml.jackson.core:jackson-databind**
FasterXML jackson-databind before 2.8.11.1 and 2.9.x before 2.9.5 allows unauthenticated remote code execution because of an incomplete fix for the CVE-2017-7525 deserialization flaw. This is exploitable by sending maliciously crafted JSON input to the readValue method of the ObjectMapper, bypassing a blacklist that is ineffective if the c3p0 libraries are available in the classpath.

Affected versions: < 2.8.11.1

**High severity vulnerability that affects com.fasterxml.jackson.core:jackson-databind**
A deserialization flaw was discovered in the jackson-databind in versions before 2.8.10 and 2.9.1, which could allow an unauthenticated user to perform code execution by sending the maliciously crafted input to the readValue method of the ObjectMapper. This issue extends the previous flaw CVE-2017-7525 by blacklisting more classes that could be used maliciously.

Affected versions: < 2.8.11

**High severity vulnerability that affects com.fasterxml.jackson.core:jackson-databind**
FasterXML jackson-databind through 2.8.10 and 2.9.x through 2.9.3 allows unauthenticated remote code execution because of an incomplete fix for the CVE-2017-7525 deserialization flaw. This is exploitable by sending maliciously crafted JSON input to the readValue method of the ObjectMapper, bypassing a blacklist that is ineffective if the Spring libraries are available in the classpath.

Affected versions: < 2.8.11

### CODE REVIEW

File: extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/LegacyKafkaIndexTaskRunner.java [hunk 1]
```diff
@@ -307,8 +307,8 @@ private TaskStatus runInternal(TaskToolbox toolbox) throws Exception
         final SeekableStreamEndSequenceNumbers<Integer, Long> restoredNextPartitions = toolbox.getObjectMapper().convertValue(
             restoredMetadataMap.get(METADATA_NEXT_PARTITIONS),
             toolbox.getObjectMapper().getTypeFactory().constructParametrizedType(
```
Comment (Fokko) [inline extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/LegacyKafkaIndexTaskRunner.java]: Curious how this ever worked. It is assigned to a `SeekableStreamEndSequenceNumbers`, but Jackson expects a `SeekableStreamStartSequenceNumbers`.

File: extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/KafkaIOConfigTest.java [hunk 2]
```diff
@@ -51,7 +50,7 @@
   public KafkaIOConfigTest()
   {
     mapper = new DefaultObjectMapper();
-    mapper.registerModules((Iterable<Module>) new KafkaIndexTaskModule().getJacksonModules());
+    mapper.registerModules(new KafkaIndexTaskModule().getJacksonModules());
```
Comment (Fokko) [inline extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/KafkaIOConfigTest.java]: Unnecessary cast

File: pom.xml [hunk 3]
```diff
@@ -84,8 +84,7 @@
         <guice.version>4.1.0</guice.version>
         <jetty.version>9.4.10.v20180503</jetty.version>
         <jersey.version>1.19.3</jersey.version>
-        <!-- jackson 2.7.x causes injection error and 2.8.x can't be used because avatica is using 2.6.3 -->
```
Comment (Fokko) [inline pom.xml]: We're already at Avatica `1.12.0`, which is at Jackson 2.9.x: https://mvnrepository.com/artifact/org.apache.calcite.avatica/avatica-core/1.12.0

**Comment by Fokko** [comment]
Rebased against master

**Comment by gianm** [comment]
Jackson and Guava are two libraries that always seem to cause dependency nightmares in connection with Hadoop. I would want to test this out with various Hadoop distributions before pulling the trigger. I know it's a pain but we have been burned here before. Is there any chance you are willing to do some of these tests?

**Comment by Fokko** [comment]
I'm familiar with the pain of classpath issues in combination with Jackson, Guava etc. Currently, we don't run Druid with Hadoop, so I'm not able to test this easily. I can try to get something running in Docker, but this will take some time.

**Comment by leventov** [comment]
Related: #4267 7152

**Comment by Fokko** [comment]
Rebased onto master, just to keep up with upstream

**Comment by Fokko** [comment]
Closing this one, this has been fixed on master.

**Pull request closed without merging**